phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java

```diff
@@ -129,43 +128,48 @@ public void testIndexRebuildTask() throws Throwable {
             Task.addTask(conn.unwrap(PhoenixConnection.class), PTable.TaskType.INDEX_REBUILD,
                     TENANT1, null, viewName,
                     PTable.TaskStatus.CREATED.toString(), data, null, startTs, null, true);
-
-
             task.run();
+            waitForTaskState(conn, PTable.TaskType.INDEX_REBUILD, viewName, PTable.TaskStatus.COMPLETED);
 
             String viewIndexTableName = MetaDataUtil.getViewIndexPhysicalName(baseTable);
             ConnectionQueryServices queryServices = conn.unwrap(PhoenixConnection.class).getQueryServices();
             int count = getUtility().countRows(queryServices.getTable(Bytes.toBytes(viewIndexTableName)));
             assertTrue(count == numOfValues);
 
-
             // Remove index contents and try again
             Admin admin = queryServices.getAdmin();
             TableName tableName = TableName.valueOf(viewIndexTableName);
             admin.disableTable(tableName);
-            admin.truncateTable(tableName, false);
+            admin.truncateTable(tableName, true);
 
             data = "{IndexName:" + indexName + ", DisableBefore:true}";
 
-            // Add a new task (update status to created) to System.Task to rebuild indexes
+            // Add a new task to System.Task to rebuild indexes
+            conn.createStatement().execute("DELETE " + " FROM " + PhoenixDatabaseMetaData.SYSTEM_TASK_NAME
+                    + " WHERE TABLE_NAME ='" + viewName  + "'");
+            conn.commit();
```

Comment:
I guess we are doing this to remove any possible tasks that may already be present in SYSTEM.TASK for the `viewName`. If so, can you please add a comment here about this?

Reply:
Actually, we don't need to remove this. I realized that Lars changed connection settings to autocommit=false. When I commit after setting status to Created, I don't need to delete the task.

Reply:
@gokceni I thought the point of this was to remove any possible leftover tasks from any previous operations, so that we ensure that all future checks only correspond to tasks created via the tests. Is this not the case? I don't understand how this is related to autocommit.